### PR TITLE
Add a simplified Rosetta CI workflow

### LINKED ISSUE #7509: Add Rosetta CI workflow
### Problem

The Rosetta Github CI job was removed in release 0.92 due to the consensus nodes no longer producing account balance file. However, we've seen recent changes breaking the Rosetta all-in-one Dockerfile without getting noticed.

We should add back the CI workflow with best effort to catch these issues. 

### Solution

Based on the original Rosetta Github CI job: https://github.com/hashgraph/hedera-mirror-node/blob/v0.91.0/.github/workflows/rosetta.yml

- don't run the CI job on pull requests
- Use previewnet: query prewviewnet for its first record file, and set importer startDate to ingest from the first record file
- Only run rosetta data reconciliation check, no construction check since it may take too long to catch up from the first record file
- Remove the BDD test step
- Remove the gosec job since it's already moved to the Gradle Github CI workflow

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by xin-hedera**
**Description**:

This PR adds a simplified Rosetta CI workflow to run data only check against previewnet.

**Related issue(s)**:

Fixes #7509 

**Notes for reviewer**:

Tested in the PR branch: https://github.com/hashgraph/hedera-mirror-node/actions/runs/7551635055/job/20559170086

Another run after addressing review feedback: https://github.com/hashgraph/hedera-mirror-node/actions/runs/7558615670/job/20580551137?pr=7555

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: .github/workflows/rosetta.yml [hunk 1]
```diff
@@ -0,0 +1,102 @@
+name: Rosetta API
+
+on:
+  schedule:
+    - cron: "0 1 * * *" # Daily at 1:00 AM
+  push:
+    tags: [ v* ]
```
Comment (steven-sheehy) [inline .github/workflows/rosetta.yml:7]: We should also have this run against main to catch issues earlier.

File: .github/workflows/rosetta.yml [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    env:
+      ROSETTA_CLI_VERSION: v0.10.3
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Set Importer StartDate
+        run: |
+          # Set start date to 60 seconds before the first block timestamp in previewnet
+          firstTimestamp=$(curl -s 'https://previewnet.mirrornode.hedera.com/api/v1/blocks?limit=1&order=asc' | \
```
Comment (steven-sheehy) [inline .github/workflows/rosetta.yml:29]: Move `previewnet` to a `NETWORK` environment variable so we don't repeat it and can quickly test with other networks.

**Review by steven-sheehy — CHANGES_REQUESTED** [review]

File: .github/workflows/rosetta.yml [hunk 3]
```diff
@@ -0,0 +1,104 @@
+name: Rosetta API
+
+on:
+  schedule:
+    - cron: "0 1 * * *" # Daily at 1:00 AM
+  push:
+    branches: [ main, 7509-add-rosetta-ci-workflow ]
```
Comment (xin-hedera) [inline .github/workflows/rosetta.yml:7]: will remove the pr branch once the workflow run succeeds

**Review by steven-sheehy — APPROVED** [review]
LGTM

**Review by edwin-greene — APPROVED** [review]
Looks good

**Pull request merged**